metricflow/model/semantics/semantic_model_lookup.py

```diff
@@ -339,3 +338,37 @@ def get_element_spec_for_name(self, element_name: str) -> LinkableInstanceSpec:
             return self._entity_ref_to_spec[EntityReference(element_name=element_name)]
         else:
             raise ValueError(f"Unable to find linkable element {element_name} in manifest")
+
+    def get_agg_time_dimension_path_key_for_measure(self, measure_reference: MeasureReference) -> ElementPathKey:
+        """Get the agg time dimension associated with the measure."""
+        agg_time_dimension = self.get_agg_time_dimension_for_measure(measure_reference)
+
+        # A measure's agg_time_dimension is required to be in the same semantic model as the measure,
+        # so we can assume the same semantic model for both measure and dimension.
+        semantic_models = self.get_semantic_models_for_measure(measure_reference)
+        assert (
+            len(semantic_models) == 1
+        ), f"Expected exactly one semantic model for measure {measure_reference}, but found semantic models {semantic_models}."
+        semantic_model = semantic_models[0]
+
+        entity_link = self.resolved_primary_entity(semantic_model)
+        assert (
+            entity_link is not None
+        ), f"Expected semantic model {semantic_model} to have a primary entity since is contains dimensions, but found none."
```

Comment:
Ok, this is implicit, right? We know it contains a measure, which means it must have an agg_time_dimension, which means it has at least a time dimension.

Maybe update the message to `to have a primary entity since it has a measure requiring an agg_time_dimension`?

Regardless, micro-nit - `it` not `is`.